samtranslator/utils/resolve_actions.py

```diff
@@ -0,0 +1,58 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict
+
+
+class ResolveAction(ABC):
```

Comment:
It's a very general class, let's keep `Resolve` out of name in case we won't need it in the future

Reply:
And same stays for the file name.

Reply:
What should the name be? Ideally I would use Actions but that's already used for intrinsic functions and I don't want overlap.

Reply:
shouldn't the traverse logic in this abstract class?